src/course-checklist/ChecklistSection/hooks.jsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from 'react';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+
+import messages from './messages';
+import getFilteredChecklist from './utils/getFilteredChecklist';
+import getValidatedValue from './utils/getValidatedValue';
+
+export const useChecklistState = ({ data, dataList }) => {
+  const [checklistState, setChecklistState] = useState({
+    checks: [],
+    totalCompletedChecks: 0,
+    values: {},
+  });
+
+  const updateChecklistState = () => {
+    if (Object.keys(data).length > 0) {
```

Comment:
I evaluate business logic that would have an elevated priority to test here as everything below that calculates values. If I was trying to test the most important 5% of the code added in this PR, I would exclude checking that `setState` is called and anything react-specific stuff, just extract the value calculations into a function and run a test on them.

Again, this is left up to you.